### ISSUE TITLE
Remove support for XEmacs

### ISSUE THREAD

**Issue by RayRacine** [ecb-home/ecb#11]
Maintaining dual compatibility with Emacs and XEmacs is onerous. With the last official XEmacs release  in 2009 with an associated remnant user base and given the reinvigoration of Emacs itself XEmacs support should be removed.  Doing so removes substantial conditionally twice written logic and code that eases code maintenance and improvements moving forward.

Consider this issue at this time as survey / poll status.

**Comment by alexott** [comment]
Agree! We may also need to look what Emacs versions will be supported to be able to use new features

**Comment by ware** [comment]
I've made a request on the mail list for people to register their thoughts in this issue.

**Comment by myrkraverk** [comment]
I come from the mailing list.  I use XEmacs 21.4.

That said, I don't have strong feelings about ECB and XEmacs.  If you want to drop it, drop it.  On the other hand, I may be persuaded to help with the XEmacs support if you decide to keep it.

**Comment by tbcs** [comment]
I share the concerns regarding the additional effort for maintaining compatibility.  I was curious to see how much compat code actually is inside ECB, so I did an experiment.  After removing all XEmacs related code ECB's size was reduced by about 5%.  IMHO this is a significant reduction, given that ECB currently consists of roughly 38 KLOC.

Is there interest in going forward with removing the compat code?  Let me know if I should open a pull request.  Right now the compat code removal consists of a [single, large, squashed commit](https://github.com/tbcs/ecb/tree/remove-xemacs-compat), but I could break it down into separate steps if necessary.

Side note: [xemacs.org](http://www.xemacs.org/) is currently offline (and apparently [has been for at least 2 weeks](https://en.wikipedia.org/w/index.php?title=XEmacs&type=revision&diff=730366405&oldid=729914004)).
